core/src/components/fab/fab.scss

```diff
@@ -35,7 +35,7 @@
 // --------------------------------------------------
 
 :host(.fab-vertical-top) {
-  top: $fab-content-margin;
+  top: max(#{$fab-content-margin}, var(--ion-safe-area-top, 0px));
```

Comment:
Is there a way we can do this without `max`? `max()` is not supported in older versions of Firefox/Chrome: https://caniuse.com/#search=max()